batch/batch/google_storage.py

```diff
@@ -34,19 +34,24 @@ def __init__(self, blocking_pool, *, project=None, credentials=None):
         self._wrapped_delete_gs_files = self._wrap_network_call(GCS._delete_gs_files)
 
     async def write_gs_file(self, uri, string, *args, **kwargs):
-        return await self._wrapped_write_gs_file(self, uri, string, *args, **kwargs)
+        return await retry_transient_errors(self._wrapped_write_gs_file,
+                                            self, uri, string, *args, **kwargs)
 
     async def read_gs_file(self, uri, *args, **kwargs):
-        return await self._wrapped_read_gs_file(self, uri, *args, **kwargs)
+        return await retry_transient_errors(self._wrapped_read_gs_file,
+                                            self, uri, *args, **kwargs)
 
     async def read_binary_gs_file(self, uri, *args, **kwargs):
-        return await self._wrapped_read_binary_gs_file(self, uri, *args, **kwargs)
+        return await retry_transient_errors(self._wrapped_read_binary_gs_file,
+                                            self, uri, *args, **kwargs)
 
     async def delete_gs_file(self, uri):
-        return await self._wrapped_delete_gs_file(self, uri)
+        return await retry_transient_errors(self._wrapped_delete_gs_file,
+                                            self, uri)
 
     async def delete_gs_files(self, uri_prefix):
```

Comment:
can you verify that deleting a URI and a URI prefix that is already deleted does not error?

Reply:
It errors in the first case, but I added a catch in the second case in case files get deleted after listing the files in the directory.